lib/idn.c

```diff
@@ -181,6 +181,18 @@ static CURLcode idn_decode(const char *input, char **output)
     result = CURLE_NOT_BUILT_IN;
 #elif defined(USE_WIN32_IDN)
   result = win32_idn_to_ascii(input, &decoded);
+#elif defined(HAVE_NETDB_H) && defined(__APPLE__)
+    /* let MacOS/iOS do this as part of DNS service */
+    struct hostent *p = gethostbyname(input);
+    if(p) {
+      if(p->h_name && (p->h_name[0])) {
+        /* host name is idn encoded by MacOS */
+        char *name = strdup(p->h_name);
+        if(name) {
+          decoded = name;
+        }
+      }
+    }
```

Comment:
I see three if() calls that I presume all should set an error code if they fail?

Reply:
Maybe worst of all: this requires that the name resolves. It can't convert just an arbitrary host name...!

Reply:
Maybe I should add some context. We had clients complaining that domains with umlauts didn't work. libidn isn't preinstalled on macOS or iOS by Apple. And we didn't want to carry the libidn2 library with every app to just get occasionally the domain resolved.

Reply:
I understand, but that does not change my comments.